packages/ui/src/fontManager.ts

```diff
@@ -0,0 +1,174 @@
+        }
+    }
+
+    private _overrideCorePlayerFontStyles(fontFamily: string): void {
+        const fontCss = `.kaltura-player-container {
+                font-family: inherit;
+            }         
+            .playkit-player {
+                font-family: ${fontFamily};
+            }
+            .playkit-player .playkit-player-gui  {
+                font-family: ${fontFamily};
+            }
+            button, textarea {
+                font-family: inherit;
+            }`;
+        const lastHeadChild = (document.head || document.getElementsByTagName("head")[0])
+            .lastElementChild;
+        const style = document.createElement("style");
+        //adding as last child
+        lastHeadChild.parentNode.insertBefore(style, lastHeadChild.nextSibling);
+        style.appendChild(document.createTextNode(fontCss));
+    }
+
+    private _checkFontExistence(fontName): boolean {
+        // creating an in-memory Canvas element
+        const canvas = document.createElement("canvas");
+        const context = canvas.getContext("2d");
+        // the text whose final pixel size we want to measure
+        const text = "abcdefghiiiiiiiiijklmnopqrstuvwwwwwwwwwwxyz0123456789";
+        // baseline font
+        context.font = `72px monospace`;
+        // size of the baseline text
+        const baselineSize = context.measureText(text).width;
+        // specifying the font whose existence we want to check
+        context.font = `72px ${fontName},monospace`;
+        // checking the size of the font we want to check
+        const newSize = context.measureText(text).width;
+        // If the size of the two text instances is the same, the font does not exist
+        // because it is being rendered using the same font
+        return newSize !== baselineSize;
+    }
+
+    private _downloadAndCacheFont(url: string, fontName: string): Promise<string | null> {
+        return axios.get(url).then((result: any) => {
+            if (result.data && typeof result.data === "string") {
+                logger.info(`font ${fontName} was downloaded successfully`, {
+                    method: "_downloadAndCacheFont"
+                });
+                this._saveFontToLocalStorage(`${FontKeyPrefix}${fontName}`, result.data);
+                return result.data;
+            } else {
+                logger.error(`failed to downloaded font ${fontName}`, {
+                    method: "_downloadAndCacheFont"
+                });
+                return null;
+            }
+        });
+    }
+
+    private _injectFontRawStyle(fontValue: string): void {
+        const style = document.createElement("style");
+        style.innerHTML = fontValue;
+        (document.head || document.getElementsByTagName("head")[0]).appendChild(style);
+        logger.info(`font raw style was injected`, {
+            method: "_injectFontRawStyle"
+        });
+    }
+
+    private _loadFontFromLocalStorage(fontKey: string): string | null {
+        try {
```

Comment:
you must check `localStorage` feature existence otherwise you will fail the flow even if font was download correctly.